### PR TITLE
more docs about diagnostics in release mode

### PR DESCRIPTION
**Pull request by dnfield**
Discussion at https://github.com/flutter/flutter/issues/56537

No tests - documentation only change.

@mraleph fyi

### CODE REVIEW

File: packages/flutter/lib/src/foundation/diagnostics.dart [hunk 1]
```diff
@@ -22,6 +22,9 @@ import 'object.dart';
 /// Trees of Flutter diagnostics can be very large so filtering the diagnostics
 /// shown matters. Typically filtering to only show diagnostics with at least
 /// level [debug] is appropriate.
+///
+/// In release mode, this level may not have any effect, as diagnostics in
+/// release mode are compatected or truncated to reduce binary size.
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/diagnostics.dart:27]: What's compatected?

Comment (goderbauer) [reply packages/flutter/lib/src/foundation/diagnostics.dart:27]: Oh, you probably meant compacted?

**Review by goderbauer — APPROVED** [review]
LGTM after typo is fixed.

File: packages/flutter/lib/src/foundation/diagnostics.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter/lib/src/foundation/diagnostics.dart]: Yeah, fixed

**Comment by dnfield** [comment]
Not waiting for CI since this is a doc only change that doesn't event introduce any links etc.

**Pull request merged**